src/components/graphs/ScatterGraph.vue

```diff
@@ -0,0 +1,499 @@
+    trendLine = chartGroup
+      .append('line')
+      .attr('x1', xScale(regression.x1))
+      .attr('y1', yScale(regression.y1))
+      // Start collapsed at (x1, y1)
+      .attr('x2', xScale(regression.x1))
+      .attr('y2', yScale(regression.y1))
+      .style('stroke', '#8b5cf6')
+      .style('stroke-width', 3)
+      .style('stroke-dasharray', '20,5')
+      .style('opacity', 0); // Start invisible
+  }
+
+  // Store references for later animation
+  chartElements = {
+    container,
+    chartGroup,
+    tooltip,
+    circles,
+    path,
+    trendLine,
+  };
+
+  loading.value = false;
+  chartRendered.value = true;
+};
+
+const animateDataElements = (): void => {
+  if (!chartElements || hasAnimated.value) return;
+
+  const { circles, path, trendLine } = chartElements;
+
+  // Animate the line
+  const totalLength = path.node()?.getTotalLength() || 0;
+  path
+    .attr('stroke-dasharray', totalLength + ' ' + totalLength)
+    .attr('stroke-dashoffset', totalLength)
+    .style('opacity', 1)
+    .transition()
+    .duration(props.animationDuration)
+    .ease(d3.easeLinear)
+    .attr('stroke-dashoffset', 0);
+
+  // Animate circles
+  circles
+    .style('opacity', 1)
+    .transition()
+    .duration(props.animationDuration)
+    .delay((d: DataPoint, i: number) => i * 100)
+    .attr('r', 6)
+    .ease(d3.easeBackOut);
+
+  // Animate trend line
+  if (trendLine) {
+    const regression = calculateLinearRegression(sortedData.value);
+    trendLine
+      .style('opacity', 0.7)
+      .transition()
+      .duration(props.animationDuration)
+      .delay(props.animationDuration * 0.2)
+      .ease(d3.easeQuadOut)
+      .attr('x2', xScale(regression.x2))
+      .attr('y2', yScale(regression.y2));
+  }
+
+  hasAnimated.value = true;
+};
+
+const calculateLinearRegression = (data: DataPoint[]): RegressionLine => {
+  const n = data.length;
+  const sumX = data.reduce((sum, d) => sum + d.year, 0);
+  const sumY = data.reduce((sum, d) => sum + d.value, 0);
+  const sumXY = data.reduce((sum, d) => sum + d.year * d.value, 0);
+  const sumXX = data.reduce((sum, d) => sum + d.year * d.year, 0);
+
+  const slope = (n * sumXY - sumX * sumY) / (n * sumXX - sumX * sumX);
+  const intercept = (sumY - slope * sumX) / n;
+
+  const xMin = Math.min(...data.map((d) => d.year));
+  const xMax = Math.max(...data.map((d) => d.year));
+
+  return {
+    x1: xMin,
+    y1: slope * xMin + intercept,
+    x2: xMax,
+    y2: slope * xMax + intercept,
+  };
+};
+
+let intersectionObserver: IntersectionObserver | undefined;
+
+onMounted(() => {
+  // Render chart structure immediately
+  renderChartStructure();
+
+  // Set up intersection observer for data animation
+  intersectionObserver = new IntersectionObserver(
+    (entries) => {
+      entries.forEach((entry) => {
+        if (
+          entry.isIntersecting &&
+          !hasAnimated.value &&
+          entry.intersectionRatio >= 0.5 &&
```

Comment:
What is the `0.5` here? Similar to colors, in general when you have a number or string being used right away, it's worth considering if it should be a constant with a clear name, which you can then document